src/Generator/InterfaceGenerator.php

```diff
@@ -0,0 +1,738 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Code\Generator;
+
+use Zend\Code\Reflection\InterfaceReflection;
+use Zend\Code\Generator\Exception\InvalidArgumentException;
+
+/**
+ * Interface generator
+ *
+ * A generator used to generate PHP interfaces.
+ *
+ * Class InterfaceGenerator
+ * @package   Zend\Code\Generator
+ * @author    Daan Biesterbos <daanbiesterbos@gmail.com>
+ * @filesource
+ */
+class InterfaceGenerator extends AbstractGenerator
+{
+    const OBJECT_TYPE = 'interface';
+
+    /**
+     * @var FileGenerator
+     */
+    protected $containingFileGenerator = null;
+
+    /**
+     * @var string
+     */
+    protected $namespaceName = null;
+
+    /**
+     * @var DocBlockGenerator
+     */
+    protected $docBlock = null;
+
+    /**
+     * @var string
+     */
+    protected $name = null;
+
+    /**
+     * @var bool
+     */
+    protected $flags = 0x00;
+
+    /**
+     * @var array Array of string names
+     */
+    protected $extendedInterfaces = array();
```

Comment:
@DaanBiesterbos Recently the rule has been applied https://github.com/zendframework/zend-code/commit/b1ddef7792190a6f8f07d2d5607a722bfc00e8dc